cdmtaskservice/mongo.py

```diff
@@ -219,6 +220,32 @@ def _doc_to_job(self, doc: dict[str, Any], as_admin: bool = False
         doc = self._clean_doc(doc)
         return models.AdminJobDetails(**doc) if as_admin else models.Job(**doc)
 
+    async def list_jobs(self, user: str | None = None, limit: int = 1000
+    ) -> list[models.JobPreview]:
+        """
+        List jobs.
+        
+        user - filter jobs by user.
+        limit - the maximum number of jobs to return.
+        """
+        # TODO LISTJOBS filter by state
+        # TODO LISTJOBS filter by time
+        query = {models.FLD_JOB_USER: user} if user else {}
+        # drop the potentially large fields
+        project = {
+            models.FLD_JOB_OUTPUTS: 0,
+            f"{models.FLD_JOB_JOB_INPUT}.{models.FLD_JOB_INPUT_INPUT_FILES}": 0
+        }
+        sort = [(_FLD_UPDATE_TIME, DESCENDING)]
+        jobs = []
+        async for j in self._col_jobs.find(
+            query, project
+            ).sort(sort
+            ).limit(_check_num(limit, "limit")
+        ):
```

Comment:
Do we need to validate `limit` (limit < 1000) before passing it to mongo? Because we already did that in the caller function?

Reply:
No, the mongo wrapper should do what the business layer tells it to do. If it tells it to pull all the data, that's what it should do. It's the business layer's responsibility to check for sane inputs